site/app/libraries/database/DatabaseQueries.php

```diff
@@ -7478,7 +7478,20 @@ public function getUserGroups(string $user_id): array {
         return $this->submitty_db->rows();
     }
 
+    public function getGroupForCourse(string $semester, string $course): string {
+        $this->submitty_db->query(
+            'SELECT group_name FROM courses WHERE semester = ? AND course = ?',
+            [$semester, $course]
+        );
+        return count($this->submitty_db->rows()) === 1 ? $this->submitty_db->rows()[0]['group_name'] : '';
+    }
+
     public function getOtherCoursesWithSameGroup(string $semester, string $course): array {
+        // some courses are archived and assigned the group "root" on production.  We want to exclude these.
+        if ($this->getGroupForCourse($semester, $course) === 'root') {
```

Comment:
if `getGroupForCourse` returns an empty string, it can mean that either the DB is corrupted and there are multiple rows with the same semester and course, or that it could not find the semester and course you passed to it. To cover both of those cases, would it not be better to also check on line 7491 that the return is not `''` and handle the error in some way?

Reply:
@cjreed121 Does line 7491 in `getOtherCoursesWithSameGroup` where we check for the group name being equal to 'root' affect the expected behavior for this function in other places this is used on Submitty (like on the page where instructors can create new courses)?

Reply:
This would not affect my page. I use the getUserGroups function. I'll go ahead and modify that to what I need in my PR to avoid seeing root as an available group. Also, rather than having the getGroupForCourse function you could just modify the SQL in this function to never select anything that has the root group.

Reply:
Do you have suggestions on how that could be done?  I wasn't sure how to do that in SQL and I figured that the other function could be useful at some point so I went ahead and added it but it might be better to write it in one query.

Reply:
Add this to the end of the query `AND c1.group_name != 'root'`. That will return 0 rows if the current course is root group.